### ISSUE TITLE
NF_CreateTextLayer: Using the same font on two screens causes on one screen to be printed in both screens.

### ISSUE THREAD

**Issue by ghost** [knightfox75/nds_nflib#13]
lets take look at A:

```
    NF_LoadTextFont("fonts/default","Default",256,256,0);
    NF_CreateTextLayer(0,0,0,"Default");
    NF_CreateTextLayer(1,0,0,"Default");

    NF_WriteText(0,0,0,0,"Made with NFlib 1.1.3");
    NF_WriteText(1,0,0,3,"Made with BlocksDS");
    NF_UpdateTextLayers();

    while(1){
        swiWaitForVBlank();
    }
```

![desmume-screenshot-0](https://github.com/knightfox75/nds_nflib/assets/172942173/e94ed702-c6c1-4a9d-91ea-fc68906f9992)
now lets take a look at B:
```
    NF_LoadTextFont("fonts/default","Default",256,256,0);
    NF_LoadTextFont("fonts/default","sDefault",256,256,0);
    NF_CreateTextLayer(0,0,0,"Default");
    NF_CreateTextLayer(1,0,0,"sDefault");

    NF_WriteText(0,0,0,0,"Made with NFlib 1.1.3");
    NF_WriteText(1,0,0,3,"Made with BlocksDS");
    NF_UpdateTextLayers();

    while(1){
        swiWaitForVBlank();
    }
```
![desmume-screenshot-1](https://github.com/knightfox75/nds_nflib/assets/172942173/cb62ee15-789e-4c24-80fe-c96c68a054a4)

i dont know whats causing this issue, but if i had to guess, its probably this line of code https://github.com/knightfox75/nds_nflib/blob/master/source/nf_text.c#L133

**Comment by ghost** [comment]
fundemental flaw with nflib i don't think it can be fixed :/ 

explanation: text fonts are basically just tiled bgs that are loaded and it's tiles are changed with text, using the same font is basically using the same bg and they first get written to ram and then updated to vram meaning both are messed up :)

**Comment by EthGaming101** [comment]
Is there any issue with loading the same text font, but giving it a different name? The text example shows something that might be the solution to that.

**Comment by AntonioND** [comment]
That should work, yes.

**Comment by ghost** [comment]
> Is there any issue with loading the same text font, but giving it a different name? The text example shows something that might be the solution to that.

no, it works nicely, it just kind of sucks to have to load the same thing again :/

**Comment by EthGaming101** [comment]
I'm gonna assume that creating a text font also creates a tile-map in RAM, and writing text to it is just updating that internal tile-map and then writing that to the screen. In that case, the loading also doubles as tile-map allocation, if that makes you feel any better.

**Comment by ghost** [comment]
> I'm gonna assume that creating a text font also creates a tile-map in RAM, and writing text to it is just updating that internal tile-map and then writing that to the screen. In that case, the loading also doubles as tile-map allocation, if that makes you feel any better.

yeah it's a fundamental flaw with nflib
